### PR TITLE
Add documentation about JLL packages

### PR DESCRIPTION
**Pull request by giordano**

### CODE REVIEW

File: docs/src/jll.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+### ExecutableProduct
+
+An [`ExecutableProduct`](@ref) is a binary executable that can be run on the
+current platform.  If, for example, the `ExecutableProduct` has been called
+`mungify_exe`, the wrapper defines an exported function named `mungify_exe` that
+should run by the user in the following way:
+
+```julia
+mungify_exe() do exe
+    run(`exe $arguments`)
```
Comment (staticfloat) [inline docs/src/jll.md:152]:
```suggestion
    run(`$exe $arguments`)
```

File: docs/src/jll.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+with the
+[`do`-block](https://docs.julialang.org/en/v1/manual/functions/#Do-Block-Syntax-for-Function-Arguments-1)
+syntax you are defining the name of the variable that will be used to actually
+call the binary with
+[`run`](https://docs.julialang.org/en/v1/base/base/#Base.run).
+
+A common point of confusion about `ExecutableProduct`s in JLL packages is why
+this function is needed: while in principle you could directly run the
+executable directly by using its absolute path in `run`, this wrapper function
+ensures that the executable will find the shared library it needs while running.
```
Comment (staticfloat) [inline docs/src/jll.md:166]:
```suggestion
ensures that the executable will find all shared libraries it needs while running.
```

File: docs/src/jll.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+The rest of the code in the wrappers is specific to each of the products of the
+JLL package and detailed below.  If you want to see a concrete example of a
+package providing all the main three products, have a look at
+[`Fontconfig_jll.jl`](https://github.com/JuliaBinaryWrappers/Fontconfig_jll.jl/tree/785936d816d1ae65c2a6648f3a6acbfd72535e36).
+
+### LibraryProduct
+
+A [`LibraryProduct`](@ref) is a shared library that can be
+[`ccall`](https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/)ed
+from Julia.  Assuming that the product is called `libdataproc`, te wrapper
```
Comment (staticfloat) [inline docs/src/jll.md:129]:
```suggestion
from Julia.  Assuming that the product is called `libdataproc`, the wrapper
```

File: docs/src/jll.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+manager](https://julialang.github.io/Pkg.jl/v1/) in the REPL with
+```
+]add NAME_jll
+```
+and then loaded with
+```
+using NAME_jll
+```
+However, most users will not ever need to do these steps on their own, JLL
+packages are usually only used as dependency of packages wrapping binary
```
Comment (staticfloat) [inline docs/src/jll.md:17]:
```suggestion
packages are usually only used as dependencies of packages wrapping binary
```

**Comment by giordano** [comment]
Should we add a link to this doc page in the `README.md` of the JLL packages?

**Pull request merged**